### PR TITLE
Live: RunStream improvements – backoff, better layout, tests

### PR DESCRIPTION
**Pull request by FZambia**
**What this PR does / why we need it**:

This pr adds several improvements to managing RunStream calls:

* backoff reconnect in case of error (though it's now based on assumption that stream closed with error fast enough). I suppose this is useful anyway to prevent busy loop re-establishment.
* better layout, move manager to `runstream` package
* add tests for clean close of `nil` error and restart on non-nil error
* use `DoAndReturn` in mocks (`Do` does not really returned a desired values to the caller).

### CODE REVIEW

File: pkg/services/live/runstream/manager.go [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+		)
+		if err != nil {
+			if errors.Is(ctx.Err(), context.Canceled) {
+				logger.Debug("Stream cleanly finished", "path", sr.Path)
+				return
+			}
+			// Best effort to cool down re-establishment process. We don't have a
+			// nice way to understand whether we really need to wait here - so relying
+			// on RunStream duration time.
+			if time.Since(startTime) < streamDurationThreshold {
```
Comment (FZambia) [inline pkg/services/live/runstream/manager.go:175]: this is where introduced backoff starts - sorry for such a pr, I better move code between packages in separate pr...

**Review by ryantxu — APPROVED** [review]
